src/inabox/inabox-viewport.js

```diff
@@ -81,11 +85,21 @@ export class ViewportBindingInabox {
     // 2) broadcast the request
     this.iframeClient_.setHostWindow(win.top);
 
+    /** @private {boolean} */
+    this.visibilityV2Enabled_ =
+        nativeIntersectionObserverSupported(win) &&
+            isExperimentOn(win, 'visibility-v2');
+
     dev().fine(TAG, 'initialized inabox viewport');
   }
 
   /** @override */
   connect() {
+    if (this.visibilityV2Enabled_) {
+      // Visibility V2 uses native IntersectionObserver, no position data needed
```

Comment:
The intersection is not calculated by AMP runtime, so no re-measure needs to happen:
https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/0.1/visibility-impl.js#L353